cache/lib/cache/s3_download_worker.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Cache.S3DownloadWorker do
+  @moduledoc """
+  Oban worker for downloading CAS artifacts from S3 to disk.
+  """
+
+  use Oban.Worker, queue: :s3_downloads, max_attempts: 3
+
+  require Logger
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"key" => key}}) do
+    Logger.info("Starting S3 download for artifact: #{key}")
+
+    if Cache.S3.exists?(key) do
+      local_path = Cache.Disk.artifact_path(key)
+
+      case download_from_s3(key, local_path) do
+        :ok ->
+          Logger.info("Successfully downloaded artifact from S3: #{key}")
+          :ok
+
+        {:error, reason} ->
+          Appsignal.send_error(%RuntimeError{message: "Failed to download artifact from S3"}, %{
+            key: key,
+            reason: reason
+          })
+
+          {:error, reason}
+      end
```

Comment:
```suggestion
      :ok = download_from_s3(key, local_path) do
        Logger.info("Successfully downloaded artifact from S3: #{key}")
          :ok
```

wouldn't we get the appsignal error reporting even when not explicitly handling the error? Or is this better because we can explicitly insert the key and error message?